### PR TITLE
GitHub Enterprise authentication for add-pr-body interceptor

### PR DESCRIPTION
**Pull request by jacobhjkim**
# Changes

`add-pr-body` interceptor only works with the public GitHub. This PR simply adds the GitHub OAuth authentication feature to the `add-pr-body` interceptor.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: tekton/ci/interceptors/add-pr-body/cmd/add-pr-body/main.go [hunk 1]
```diff
@@ -38,11 +39,18 @@ const (
 )
 
 func main() {
-	http.HandleFunc("/", makeAddPRBodyHandler(getPrBody))
+	http.HandleFunc("/", makeAddPRBodyHandler(getPrBody, getGitHubAuth("GITHUB_OAUTH_SECRET", "")))
 	log.Fatal(http.ListenAndServe(fmt.Sprintf(":%d", 8080), nil))
 }
 
-func makeAddPRBodyHandler(urlFetcherDecoder urlToMap) http.HandlerFunc {
+func getGitHubAuth(key, fallback string) string {
```
Comment (afrittoli) [inline tekton/ci/interceptors/add-pr-body/cmd/add-pr-body/main.go:46]: NIT: does the fallback need to be configurable?

**Comment by afrittoli** [comment]
Please ignore the "try-out-prow-plus-tekton" - it was an experiment, it's not a required check so it shouldn't cause issues.

**Comment by jacobhjkim** [comment]
`plumbing-golang-lint` was triggered and the job ran successfully. However, `github-check-done` didn't work properly. @afrittoli

**Review by afrittoli — COMMENTED** [review]
Nice, thanks!
/approve

**Comment by afrittoli** [comment]
We do not have CD support yet for this interceptor, so it will have to be re-deployed manually once this PR is merged.
We would not really use auth in Tekton infra, but it would be good to keep the binary up to date.

**Review by vdemeester — COMMENTED** [review]
/lgtm

**Pull request merged**